doc/OnlineDocs/tests/test_examples.py

```diff
@@ -1,269 +1,68 @@
-solver_available = {}
-package_available = {}
-
-only_book_tests = set(['Test_nonlinear_ch', 'Test_scripts_ch'])
-
+from pyomo.common.dependencies import attempt_import
+from pyomo.common.fileutils import this_file_dir
+import pyomo.environ as pyo
 
-def _check_available(name):
-    from pyomo.opt.base.solvers import check_available_solvers
 
-    return bool(check_available_solvers(name))
+parameterized, param_available = attempt_import('parameterized')
+if not param_available:
+    raise unittest.SkipTest('Parameterized is not available.')
 
+# Needed for testing (switches the matplotlib backend):
+from pyomo.common.dependencies import matplotlib_available
 
-def check_skip(tfname_, name):
-    #
-    # Skip if YAML isn't installed
-    #
-    if not yaml_available:
-        return "YAML is not available"
-    #
-    # Initialize the availability data
-    #
-    if len(solver_available) == 0:
-        for tf_ in solver_dependencies:
-            for n_ in solver_dependencies[tf_]:
-                solver_ = solver_dependencies[tf_][n_]
-                if not solver_ in solver_available:
-                    solver_available[solver_] = _check_available(solver_)
-        for tf_ in package_dependencies:
-            for n_ in package_dependencies[tf_]:
-                packages_ = package_dependencies[tf_][n_]
-                for package_ in packages_:
-                    if not package_ in package_available:
-                        try:
-                            __import__(package_)
-                            package_available[package_] = True
-                        except:
-                            package_available[package_] = False
-    #
-    # Return a boolean if the test should be skipped
-    #
-    if tfname_ in solver_dependencies:
-        if (
-            name in solver_dependencies[tfname_]
-            and not solver_available[solver_dependencies[tfname_][name]]
-        ):
-            # Skip the test because a solver is not available
-            # print('Skipping %s because of missing solver' %(name))
-            return 'Solver "%s" is not available' % (
-                solver_dependencies[tfname_][name],
-            )
-    if tfname_ in package_dependencies:
-        if name in package_dependencies[tfname_]:
-            packages_ = package_dependencies[tfname_][name]
-            if not all([package_available[i] for i in packages_]):
-                # Skip the test because a package is not available
-                # print('Skipping %s because of missing package' %(name))
-                _missing = []
-                for i in packages_:
-                    if not package_available[i]:
-                        _missing.append(i)
-                return "Package%s %s %s not available" % (
-                    's' if len(_missing) > 1 else '',
-                    ", ".join(_missing),
-                    'are' if len(_missing) > 1 else 'is',
-                )
-    return False
+bool(matplotlib_available)
```

Comment:
I cannot figure out the purpose of this line. Is this not already a bool? And also, you aren't assigning this to anything, so is this line actually _doing_ anything?

Reply:
It actually *is* doing something: `attempt_import` returns an object (not a bool) for the `available` return value.  When that object is cast to a bool (usually in something like `if package_available:`), then the `attempt_import` mechanism is triggered to 1) attempt the import, and 2) resolve the `available` object to it's correct True/False value.  I blame the confusion on Black: the comment explaining this is before the import, and Black insisted on adding 2 spaces after the import to separate the import and the cast to improve confusion.